### PR TITLE
Added onnx export support for SelectColumns

### PR DESCRIPTION
**Pull request by harishsk**
Implemented ability to export ColumnSelectingEstimator to onnx.

### CODE REVIEW

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 1]
```diff
@@ -1254,6 +1254,60 @@ public void CopyColumnsOnnxTest()
...
+                Assert.Equal("Size1", outputNames[0]);
+                Assert.Equal("Shape1", outputNames[1]);
+                Assert.Equal("Thickness1", outputNames[2]);
+                Assert.Equal("Label1", outputNames[3]);
+
+                CompareSelectedScalarColumns<int>("Size", "Size1", transformedData, onnxResult);
+                CompareSelectedScalarColumns<int>("Shape", "Shape1", transformedData, onnxResult);
+                CompareSelectedScalarColumns<int>("Thickness", "Thickness1", transformedData, onnxResult);
+                CompareSelectedScalarColumns<bool>("Label", "Label1", transformedData, onnxResult);
+            }
```
Comment (codemzs) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: for linux/centos platforms where ORT is not supported can we have a baseline test to compare against?

File: src/Microsoft.ML.Data/Transforms/ColumnSelecting.cs [hunk 2]
```diff
@@ -725,6 +727,22 @@ DataViewRow IRowToRowMapper.GetRow(DataViewRow input, IEnumerable<DataViewSchema
...
+                var outputToInputMap = _mapper.OutputToInputMap;
+                for(int i = 0; i < outputToInputMap.Length; i++)
+                {
+                    var srcCol = InputSchema[outputToInputMap[i]];
+                    var dstCol = OutputSchema[i];
+                    var srcVariable = ctx.GetVariableName(srcCol.Name);
+                    var dstVariable = ctx.AddIntermediateVariable(dstCol.Type, dstCol.Name, true);
+                    string opType = "Identity";
+                    ctx.CreateNode(opType, srcVariable, dstVariable, ctx.GetNodeName(opType), "");
+                }
```
Comment (codemzs) [inline src/Microsoft.ML.Data/Transforms/ColumnSelecting.cs]: Where are you dropping remaining columns?

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 3]
```diff
@@ -1254,6 +1254,60 @@ public void CopyColumnsOnnxTest()
...
+
+                Assert.Equal("Size1", outputNames[0]);
+                Assert.Equal("Shape1", outputNames[1]);
+                Assert.Equal("Thickness1", outputNames[2]);
+                Assert.Equal("Label1", outputNames[3]);
+
+                CompareSelectedScalarColumns<int>("Size", "Size1", transformedData, onnxResult);
+                CompareSelectedScalarColumns<int>("Shape", "Shape1", transformedData, onnxResult);
+                CompareSelectedScalarColumns<int>("Thickness", "Thickness1", transformedData, onnxResult);
+                CompareSelectedScalarColumns<bool>("Label", "Label1", transformedData, onnxResult);
```
Comment (codemzs) [inline test/Microsoft.ML.Tests/OnnxConversionTest.cs]: I think we want to make sure the columns you have not selected are not present in the output schema anymore, that is the point of select column transform.

Comment (harishsk) [reply test/Microsoft.ML.Tests/OnnxConversionTest.cs]: Fixed to verify that the output schema contains only the selected columns

---
In reply to: [361511408](https://github.com/dotnet/machinelearning/pull/4590#discussion_r361511408) [](ancestors = 361511408)

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs (hunk 1 above)
Comment (harishsk) [reply test/Microsoft.ML.Tests/OnnxConversionTest.cs]: Added baseline comparison

---
In reply to: [361510683](https://github.com/dotnet/machinelearning/pull/4590#discussion_r361510683) [](ancestors = 361510683)

File: src/Microsoft.ML.Data/Transforms/ColumnSelecting.cs (hunk 2 above)
Comment (harishsk) [reply src/Microsoft.ML.Data/Transforms/ColumnSelecting.cs]: By default, only the columns that have connections are selected. So the columns not in the OutputToInputMap are automatically dropped.

But your comment made me look deeper into the output schema. I have fixed the output schema with an updated to OnnxTransformer.

---
In reply to: [361511119](https://github.com/dotnet/machinelearning/pull/4590#discussion_r361511119) [](ancestors = 361511119)

**Pull request merged**